earth_enterprise/src/fusion/autoingest/StorageManager.h

```diff
@@ -355,5 +368,30 @@ bool StorageManager<AssetType>::SaveDirtyToDotNew(
   return true;
 }
 
+template<class AssetType>
+bool StorageManager<AssetType>::DetermineIfPrune() {
+  std::lock_guard<std::recursive_mutex> lock(storageMutex);
+  uint32 cacheSize = CacheSize();
+  uint32 dirtySize = DirtySize();
+  uint32 oriNum;
+  uint32 newNum;
+  uint32 diff;
+  float percent;
+
+  if (cacheSize > dirtySize) {
+    oriNum = cacheSize;
+    newNum = dirtySize;
+  }
+  else {
+    oriNum = dirtySize;
+    newNum = cacheSize;
+  }
+  if (oriNum == newNum || oriNum == 0 || newNum == 0) { return false; }
+  diff = oriNum - newNum;
+  percent = ((float)diff / (float)oriNum) * 100;
+  notify(NFY_WARN, "Percent Diff: %f", percent);
+  return (percent > maxSizeDiff);
```

Comment:
I think this code is correct but it's confusing to me to write it this way. I think you can replace most of the body of this function with `return (dirtySize < cacheSize * maxSizeDiff);` Then you only prune if the dirty map isn't taking up a large portion of the cache. You could equivalently use `return (dirtySize < cacheSize * (1 - maxSizeDiff));` if you prefer to specify the sizes differently.

Also, I'd recommend changing `maxSizeDiff` to `maxPurgePercent` or something a little more descriptive.